tests,vm-virtio,vmm: Use 'socket' for all CLI/API parameters

likebreath opened the pull request:
This patch unifies the inconsistent uses of 'socket' and 'sock' from our
CLI/API parameters.

Fixes: #1091

Signed-off-by: Bo Chen <chen.bo@intel.com>

rbradford commented:
This is great. For the vhost_user_fs daemon binary please make it accept both the old and new names giving a deprecation warning for the old.

This is because I think other folks may be using that binary.

We can complete the rename after the release by removing then.

rbradford commented on src/bin/vhost_user_fs.rs:
```diff
@@ -295,8 +295,8 @@ fn main() {
-            Arg::with_name("sock")
-                .long("sock")
+            Arg::with_name("socket")
+                .long("socket")
```
This is the only one where I think we should temporarily support both parameters.

likebreath replied:
Thanks for the comments and pointers. I just updated the PR, and that seems to serve the purpose. Please let me know if there is better ways to do it. :)

sboeuf commented on vmm/src/config.rs:
```diff
@@ -1749,27 +1749,27 @@ mod tests {
+            FsConfig::parse("tag=mytag,socket=/tmp/socket,dax=on")?,
             FsConfig {
-                sock: PathBuf::from("/tmp/sock"),
+                socket: PathBuf::from("/tmp/socket"),
```
Ditto

sboeuf commented on vmm/src/config.rs:
```diff
@@ -1749,27 +1749,27 @@ mod tests {
         );
         assert_eq!(
-            FsConfig::parse("tag=mytag,sock=/tmp/sock,dax=on,cache_size=4G")?,
+            FsConfig::parse("tag=mytag,socket=/tmp/socket,dax=on,cache_size=4G")?,
```
Ditto

sboeuf commented on vmm/src/config.rs:
```diff
@@ -1749,27 +1749,27 @@ mod tests {
+            FsConfig::parse("tag=mytag,socket=/tmp/socket,dax=on,cache_size=4G")?,
             FsConfig {
-                sock: PathBuf::from("/tmp/sock"),
+                socket: PathBuf::from("/tmp/socket"),
```
Ditto

sboeuf commented on vmm/src/config.rs:
```diff
@@ -1749,27 +1749,27 @@ mod tests {
         );
         // Cache size without DAX is an error
-        assert!(FsConfig::parse("tag=mytag,sock=/tmp/sock,dax=off,cache_size=4G").is_err());
+        assert!(FsConfig::parse("tag=mytag,socket=/tmp/socket,dax=off,cache_size=4G").is_err());
```
Ditto

likebreath commented:
@sboeuf I have updated the changes based on your comments. I also cleaned up few places from `main.rs` and `config.rs` so that we only use "sock" (and not "socket") from file paths. For example: changes from `src/main.rs:748`. Any more advices, please let me know. Thanks!

sboeuf commented:
Thanks @likebreath

rbradford commented:
@sboeuf did @likebreath address all your comments?
